src/daemon/devnode.c

```diff
@@ -92,7 +121,7 @@ int _mknotifynode(usbdevice* kb, int notify){
         return 0;
     // Create the notification node
     int index = INDEX_OF(kb, keyboard);
-    char outpath[strlen(devpath) + 10];
+    char outpath[strlen(devpath) + 8*sizeof(char) + 2*sizeof(int)];
```

Comment:
I think you've misunderstood the code here; the original code was correct because `index` is always between `1` and `9`.

Your code is incorrect even under the assumption this invariant does not hold; the largest string a 32-bit integer can hold is `-2147483648` which is 11 characters, not 4, let alone a 64-bit int (which is rare), which has the longest string of `-9223372036854775808`, which is 20 characters long.